### ISSUE TITLE
How were the metadata .pkl files created?

### ISSUE THREAD

**Issue by Stelath** [qinzheng93/GeoTransformer#88]
Hello, I was wondering what script was used to create the metadata `.pkl` files as I haven't been able to find it anywhere in the provided code. I was hoping that you could send me the script you used to create the Kitti metadata or if you could point me in the right direction on how to make them for a dataset similar in structure to kitti that would be great! Thanks for any help!

**Comment by xcvbn123456** [comment]
> Hello, I was wondering what script was used to create the metadata `.pkl` files as I haven't been able to find it anywhere in the provided code. I was hoping that you could send me the script you used to create the Kitti metadata or if you could point me in the right direction on how to make them for a dataset similar in structure to kitti that would be great! Thanks for any help!

Have you solved this problem?

**Comment by Stelath** [comment]
We reverse engineered it but we've had consistency issues

**Comment by claireabraeuer** [comment]
How did you reverse engineer it? I'm trying to create my own `.pkl` files as well but I haven't found how they arrived at the ground truth transforms for their sequence. 

![Image](https://github.com/user-attachments/assets/bbf2e7a8-3928-4125-b4b6-7d565376551a)

**Comment by Stelath** [comment]
Don't have the code on me rn, our super computer just shut down for maintenance lol, and its been a second cause our papers pretty much done and I'm at an internship but I'll pull the file out when I get the chance in like a week.

**Comment by Stelath** [comment]
If I remember correctly after a lot of digging they didn't even make the metadata file, they used predators, which came from some other guy from 2019 or something, it works by running icp across all the points to create the ground truth (**DON'T FORGET TO DO THE CAMERA TRANSLATION** that cost us like 3 weeks lol) and we we're finally able to recreate their results by just reverse engineering the rest, in terms of picking point clouds that were roughly 10m apart, etc.

What are you working on?

**Comment by claireabraeuer** [comment]
Oh, thank you for your quick reply!
Are you referring to [this](https://github.com/prs-eth/OverlapPredator/blob/8c78f125fc58d62ad7d149adf1fed43ed54937e4/datasets/kitti.py#L105C9-L127C1) ICP refinement from the PREDATOR codebase? And regarding the camera translation do you mean that I have to use the correct `T_velo_cam` transform to go from `cam0` to the velodyne sensor, or is there some additional translation I need to consider?

I'm currently just trying to run GeoTransformer on a different sequence of frame pairs for which I have to compute the correct ground truth transforms.

**Comment by Stelath** [comment]
Yes, you are correct on both fronts, we just didn't do the `T_velo_cam` on accident; My advice: visually plot a couple of the transforms registered together after computing the metadata file just to verify they're correct.

**Comment by claireabraeuer** [comment]
Thank you so much for the advice! I'm currently running the PREDATOR's KITTI processing and I hope I'll get something close to the GeoTransformer `.pkl` transforms now. :)
